src/fmu/tools/rms/zone_mapping.py

```diff
@@ -0,0 +1,242 @@
+        return zone_numbers
+
+    def get_zone_number(self, zone_name: str):
+        # Unique zone name for each zone code
+        for code, name in self.zone_code_names.items():
+            if zone_name == name:
+                return code
+        raise ValueError(f"Unknown zone name: {zone_name}")
+
+    def get_zone_name_for_zone_number(self, zone_number: int):
+        return self.zone_code_names[zone_number]
+
+    def get_zone_name_for_zone_index(
+        self, zone_index: int, zone_name_from_param: bool = True
+    ):
+        """
+        Return zone name in zone_code table in zone parameter or
+        alternatively zone name from grid.
+        """
+        if not zone_name_from_param:
+            # Zone name from grid
+            return self.grid.zone_names[zone_index]
+        # zone name from param
+        zone_number = self.get_zone_number_for_zone_index(zone_index)
+        return self.zone_code_names[zone_number]
+
+    def get_zone_number_for_zone_index(self, zone_index: int):
+        return self.grid_zone_dict[zone_index]["zone_number"]
+
+    def get_zone_index_for_zone_number(self, zone_number: int):
+        self.validate_zone_number_for_grid(zone_number)
+        return self.grid_zone_index_dict[zone_number]
+
+    def is_zone_number_defined(self, zone_number: int):
+        zone_numbers_in_grid = self.get_zone_numbers_in_grid()
+        return zone_number in zone_numbers_in_grid
+
+    def number_of_layers_for_zone_number(self, zone_number: int):
+        self.validate_zone_number_for_grid(zone_number)
+        index = self.grid_zone_index_dict[zone_number]
+        return self.grid_zone_dict[index]["nlayers"]
+
+    def number_of_layers_for_zone_index(self, zone_index: int):
+        return self.grid_zone_dict[zone_index]["nlayers"]
+
+    def get_start_end_layer_for_zone_number(self, zone_number: int):
+        self.validate_zone_number_for_grid(zone_number)
+        index = self.grid_zone_index_dict[zone_number]
+        return self.grid_zone_dict[index]["start_layer"], self.grid_zone_dict[index][
+            "end_layer"
+        ]
+
+    def get_start_end_layer_for_zone_index(self, zone_index: int):
+        return self.grid_zone_dict[zone_index]["start_layer"], self.grid_zone_dict[
+            zone_index
+        ]["end_layer"]
+
+    def get_zone_layer_numbering(self):
+        nzones = self.get_number_of_zones_in_grid()
+        number_of_layers_per_zone = []
+        start_layers_per_zone = []
+        end_layers_per_zone = []
+        for zone_index in range(nzones):
+            number_of_layers_per_zone.append(
+                self.number_of_layers_for_zone_index(zone_index)
+            )
+            start, end = self.get_start_end_layer_for_zone_index(zone_index)
+            start_layers_per_zone.append(start)
+            end_layers_per_zone.append(end)
+        return number_of_layers_per_zone, start_layers_per_zone, end_layers_per_zone
+
+    def get_number_of_layers_per_zone(self):
```

Comment:
I don't think this method is applied anywhere. It is also overlapping with `get_zone_layer_numbering`?

Reply:
Is used in APS code, but get_zone_layer_numbering can be removed